### PR TITLE
Compatibility with phonegap 2.0 (Cordova)

### PR DESCRIPTION
**Pull request by aetchegoyen**
With this change the plugin works perfecly on phonegap 2.0

### CODE REVIEW

File: Android/DatePicker/datePickerPlugin.js [hunk 1]
```diff
@@ -43,10 +43,5 @@ if (typeof cordova !== "undefined") {
 		console.log("datePickerPlugin.js failed: " + err);
 	}
 
-	cordova.addConstructor(function() {
-		if (!window.plugins) {
-			window.plugins = {};
-		}
-		window.plugins.datePicker = new DatePicker();
-	});
+	window.datePicker = new DatePicker();
```
Comment (devgeeks) [inline Android/DatePicker/datePickerPlugin.js:46]: Why would you move the plugin to the global scope?

**Comment by devgeeks** [comment]
see comment on changed file.

File: Android/DatePicker/datePickerPlugin.js (hunk 1 above)
Comment (aetchegoyen) [inline Android/DatePicker/datePickerPlugin.js:46]: In Cordova 2.0.0 the cordova.addConstructor() method has been removed. 
But you're right about the global scope. Perhaps a good workout would be to manually create the plugins object in window if it doens't exist (yeah, still global but at least organized), and attach the plugin there. 

For example: 

if(!window.plugins) window.plugins = {};
window.plugins.datePicker = new datePicker();

**Comment by aetchegoyen** [comment]
Answered, thanks for your feedback! I'm working on another issue: if you blur the input the datepicker keeps showing up, not allowing to cancel the date picking (Android 4.1.2, Cordova 2.0), i'll let you know when i find a solution

File: Android/DatePicker/datePickerPlugin.js (hunk 1 above)
Comment (devgeeks) [inline Android/DatePicker/datePickerPlugin.js:46]: That would certainly be more along use expectations as well.

Another option would be to update plugins to the [Cordova plugin spec](https://github.com/alunny/cordova-plugin-spec).

**Comment by aetchegoyen** [comment]
Perfect, I'll prepair a new pull request according to phonegap plugin spec and considering all versions

**Comment by vdevadass** [comment]
Hi aetchegoyen, I ran into the cancel button issue you mention above. Do you have a solution for this problem? Thanks in advance!

**Comment by aetchegoyen** [comment]
Hi!
     Thanks for your feedback I couldn't touch that develpment until 
today. I saw your other email too, I'm very glad you worked it out! I'll 
check your code tomorrow first thing in the morning.
Cheers!
Lets keep in touch!
Ale
    Alejandro Etchegoyen
TE: (+54)(011) 5263-0144
GTalk: alejandro.etchegoyen@gmail.com Skype: aetchegoyen
Av. Hipólito Yrigoyen 7118 (B1828) Banfield, Buenos Aires, Argentina

**Comment by shazron** [comment]
This repo has been deprecated and will not be actively maintained - no pull requests nor any other issues will be worked on. The previous source currently exists in the DEPRECATED branch, and may be removed at a future date.
Migration path, see: https://github.com/phonegap/phonegap-plugins/blob/master/README.md

**Pull request closed without merging**